### PR TITLE
UI - Push to GitHub / SelectOptions

### PR DESCRIPTION
**Pull request by eahrold**
Companion to the Push to GitHub Feature https://github.com/micronaut-projects/micronaut-starter/pull/495

This also adds support the SelectOptions featuere https://github.com/micronaut-projects/micronaut-starter/pull/545

### CODE REVIEW

**Comment by pgressa** [comment]
HI @eahrold thanks! I'm going to test it today just for sure, so we can give it a green light. First the backend stuff needs to merged, after that this one will go in.

**Comment by pgressa** [comment]
@eahrold I tested the application and it seems ok. The error responses are being handled as well as happy scenario. 
One comment I have is that I'm not confident is the formulation of Clone to GitHub, but I believe people using this know what that means, even though we're not cloning it, but we're creating new repository for them, with generated content.


Additionally, not sure how much time you have for this, but from user perspective, it would be great to copy+paste the clone link, as now user has to navigate to github first to copy the clone link. But for sure we can do this in next iteration. The plan is to extend the feature set by github ci/cd so once we have such features available, we should offer such specific features on  clone to github action.

**Comment by eahrold** [comment]
@pgressa @graemerocher I spent some time trying to do some clipboard business in the current alert/toast panel, but none of it really worked, so I went back to the drawing board and added a `NextSteps` component that lists out all the commands, and is just a bit cleaner. 

I'm open to any verbiage changes. @JasonTypesCodes you may have some suggestions too.

File: main/src/main/js/launch/package.json [hunk 1]
```diff
@@ -16,9 +16,11 @@
   },
   "scripts": {
     "start": "react-scripts start",
+    "start-local" : "export export REACT_APP_API_URL=http://localhost:8080; export REACT_APP_SNAPSHOT_API_URL=http://localhost:8080;export CORS_ALLOWED_ORIGIN=http://localhost:3000; npm run start",
```
Comment (JasonTypesCodes) [inline main/src/main/js/launch/package.json:19]: Is export intentionally in here twice?

File: main/src/main/js/launch/src/components/NextSteps/NextSteps.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            }
+    }, [buildTool])
+
+
+    const cmdData = useMemo(()=>{
+        switch (info.type) {
+            case 'clone':
+                return {action: "Clone the repo", cmd: `git clone ${cloneUrl}`}
+            case "zip":
+                return {action: "unzip the archive", cmd: `unzip ${name}.zip`}
```
Comment (JasonTypesCodes) [inline main/src/main/js/launch/src/components/NextSteps/NextSteps.js:22]: I think this should be "Unzip" to match the casing of the other steps

**Review by JasonTypesCodes — COMMENTED** [review]
This lgtm.  Just left some minor feedback

**Comment by graemerocher** [comment]
@jameskleeh @JasonTypesCodes can this be reviewed and merged? With this we will be able to test the snapshot right?

**Comment by jameskleeh** [comment]
We can’t merge it until 2.2 is out because the functionality only works on
snapshot currently. We’ve tested it as much as we can I think.

**Comment by graemerocher** [comment]
Ah I see, I thought it would gracefully handle both APIs..

**Comment by eahrold** [comment]
> Ah I see, I thought it would gracefully handle both APIs..

I could actually get that in place pretty easily if we provided a list of "supported features" or something to that extent.  

Maybe pull down the swagger `.yml` and do some checking against that. 
Maybe we could turn on the `/info` endpoint and use the commit count, but I'm not sure how reliable that would be.
Open to any other ideas here

**Comment by pgressa** [comment]
@eahrold it works fantastic! 💯

**Pull request merged**